app/services/hmrc/mock_interface_response_service.rb

```diff
@@ -16,6 +16,7 @@ def self.call(*args)
     def initialize(hmrc_response)
       @hmrc_response = hmrc_response
       @application = @hmrc_response.legal_aid_application
+      @hmrc_response.submission_id = SecureRandom.uuid
```

Comment:
Does this save automatically?
If not, setting line 20 to 
```ruby 
@submission_id = @hmrc_response.submission_id
```

and then changing line 24 to 
```ruby
@hmrc_response.update!(response: return_json.to_json, submission_id: @submission_id)
```
Which matches the action of the HMRC::SubmissionWorker